### ISSUE TITLE
MueLu, yaml-cpp, Boost, Unit test failures

### ISSUE THREAD

**Issue by jjellio** [trilinos/Trilinos#1093]
When building with
TPLs yaml-cpp and boost, MueLu's unit tests will fail to build, because CMake will fail to find yaml-cpp files during configuration.

Should this tests be in MueLu still? Perhaps it belongs in Teuchos, or is it redundant?

packages/muelu/test/unit_tests/CMakeLists.txt has:

```
IF (TPL_ENABLE_yaml-cpp AND TPL_ENABLE_Boost)
  INCLUDE_DIRECTORIES(${yaml-cpp_INCLUDE_DIRS})
  INCLUDE_DIRECTORIES(${CMAKE_CURRENT_SOURCE_DIR}/../../src/Interface/YamlParameterList)
  APPEND_SET(SOURCES yaml/YamlParameterList.cpp)
  TRIBITS_COPY_FILES_TO_BINARY_DIR(MueLu_Yaml_cp
    SOURCE_DIR ${MueLu_SOURCE_DIR}/test/unit_tests/yaml/
    DEST_DIR ${CMAKE_CURRENT_BINARY_DIR}/yaml/
    SOURCE_FILES Match1.yaml Match1.xml Match2.yaml Match2.xml Match3.yaml Match3.xml Match4.yaml Match4.xml Arrays.yaml IllegalKeyString.yaml IntVsDouble.xml IntVsDouble.yaml MueLuConfig.yaml MueLuConfig.xml InconsistentArray.yaml input_restingHydrostatic_RK4.xml plasma_oscillation_rtc.xml
  )
ENDIF()
```

YamlParameterList lives in Teuchos now:

```
[jjellio@mutrino:~/src/Tril...nos-jjellio/packages] find -name "*YamlParameterList*"
./teuchos/comm/src/Teuchos_YamlParameterListHelpers.cpp
./teuchos/comm/src/Teuchos_YamlParameterListHelpers.hpp
./teuchos/parameterlist/src/Teuchos_YamlParameterListCoreHelpers.hpp
./teuchos/parameterlist/test/yaml/YamlParameterList.cpp

```

@trilinos/muelu @trilinos/teuchos

**Comment by nmhamster** [comment]
@jjellio what yaml-cpp are you using it has gone through a lot of changes recently? Are you on a testbed?

**Comment by jhux2** [comment]
> When building with
> TPLs yaml-cpp and boost, MueLu's unit tests will fail to build, because CMake will fail to find yaml-cpp files during configuration.
> 
> Should this tests be in MueLu still? Perhaps it belongs in Teuchos, or is it redundant?

@jjellio This looks like build logic that didn't get purged.  If you remove it, do MueLu's unit tests then build for you?

**Comment by jhux2** [comment]
@jjellio There are similar issues in ```muelu/src/Interface/CMakeLists.txt``` and ```muelu/test/interface/CMakeLists.txt```.  Could you try removing that also?

**Comment by jhux2** [comment]
@jjellio checkin test is running now

**Comment by jhux2** [comment]
@jjellio This should now be fixed with commit 8303b75.  Please reopen if you still see problems with MueLu and yaml-cpp.

**Comment by jjellio** [comment]
Yes, sorry for the delay. I 'fixed' them locally by disabling that part of the CMakeLists.txt

I just hit this, because I just turned on Yaml-cpp as part of my benchmarking efforts.. the builds locally never failed, because I do not need Boost... but my builds on the testbeds target the various  apps, so I enable many more TPLS.

I'll pull again and post back if there are problems.
